pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -385,7 +396,6 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 	firstUnhealthyOrdinal := math.MaxInt32
 	var firstUnhealthyPod *v1.Pod
 	monotonic := !allowsBurst(set)
-	minReadySeconds := getMinReadySeconds(set)
```

Comment:
consider moving scaleMaxUnavailable variable initilaization inside processReplicas

Reply:
The variable scaleMaxUnavailable should not be initialized inside the processReplicas function because the function will modify it based on the metrics (decrementing by 1 for each unhealthy instance).